tests/functional/codegen/features/decorators/test_nonreentrant.py

```diff
@@ -60,7 +60,7 @@ def protected_function2(val: String[100], do_callback: bool) -> uint256:
     return 2
 
 @external
-@nonreentrant('protect_special_value')
+@nonreentrant
 def protected_function3(val: String[100], do_callback: bool) -> uint256:
```

Comment:
I actually don't understand why this function compiles since in the `Self` interface `protected_view_fn` is defined as `view` function, but below it's implemented as a state-changing function...

Reply:
https://github.com/vyperlang/vyper/pull/3769/commits/59804e0aad29be7c26e2c9250d7ae25ea75605db